src/helpers/figures/charts/waterfall_chart.ts

```diff
@@ -0,0 +1,357 @@
+    return this.getDefinitionWithSpecificDataSets(this.dataSets, this.labelRange);
+  }
+
+  private getDefinitionWithSpecificDataSets(
+    dataSets: DataSet[],
+    labelRange: Range | undefined,
+    targetSheetId?: UID
+  ): WaterfallChartDefinition {
+    return {
+      type: "waterfall",
+      dataSetsHaveTitle: dataSets.length ? Boolean(dataSets[0].labelCell) : false,
+      background: this.background,
+      dataSets: dataSets.map((ds: DataSet) =>
+        this.getters.getRangeString(ds.dataRange, targetSheetId || this.sheetId)
+      ),
+      legendPosition: this.legendPosition,
+      verticalAxisPosition: this.verticalAxisPosition,
+      labelRange: labelRange
+        ? this.getters.getRangeString(labelRange, targetSheetId || this.sheetId)
+        : undefined,
+      title: this.title,
+      aggregated: this.aggregated,
+      showSubTotals: this.showSubTotals,
+      showConnectorLines: this.showConnectorLines,
+      positiveValuesColor: this.positiveValuesColor,
+      negativeValuesColor: this.negativeValuesColor,
+      subTotalValuesColor: this.subTotalValuesColor,
+    };
+  }
+
+  getDefinitionForExcel(): ExcelChartDefinition | undefined {
+    // TODO: implement export excel
+    return undefined;
+  }
+
+  updateRanges(applyChange: ApplyRangeChange): WaterfallChart {
+    const { dataSets, labelRange, isStale } = updateChartRangesWithDataSets(
+      this.getters,
+      applyChange,
+      this.dataSets,
+      this.labelRange
+    );
+    if (!isStale) {
+      return this;
+    }
+    const definition = this.getDefinitionWithSpecificDataSets(dataSets, labelRange);
+    return new WaterfallChart(definition, this.sheetId, this.getters);
+  }
+}
+
+function getWaterfallConfiguration(
+  chart: WaterfallChart,
+  labels: string[],
+  dataSeriesLabels: (string | undefined)[],
+  localeFormat: LocaleFormat
+): ChartConfiguration {
+  const { locale, format } = localeFormat;
+
+  const fontColor = chartFontColor(chart.background);
+  const config = getDefaultChartJsRuntime(chart, labels, fontColor, localeFormat);
+  const negativeColor = chart.negativeValuesColor || CHART_WATERFALL_NEGATIVE_COLOR;
+  const positiveColor = chart.positiveValuesColor || CHART_WATERFALL_POSITIVE_COLOR;
+  const subTotalColor = chart.subTotalValuesColor || CHART_WATERFALL_SUBTOTAL_COLOR;
+  const legend: DeepPartial<LegendOptions<"bar">> = {
+    labels: {
+      generateLabels: () => {
+        const legendValues = [
+          { text: _t("Positive values"), fontColor, fillStyle: positiveColor },
+          { text: _t("Negative values"), fontColor, fillStyle: negativeColor },
+        ];
+        if (chart.showSubTotals) {
+          legendValues.push({
+            text: _t("Subtotals"),
+            fontColor,
+            fillStyle: subTotalColor,
+          });
+        }
+        return legendValues;
+      },
+    },
+  };
+  if ((!chart.labelRange && chart.dataSets.length === 1) || chart.legendPosition === "none") {
```

Comment:
I'm not sure about this condition. Unlike other charts, the legend should be visible here even if we have one dataset and no label, no ? To have the "negative", "positive", ... display

Reply:
You're right :)